runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -429,41 +429,33 @@ J9::CodeGenerator::lowerCompressedRefs(
       }
    else
       {
-      TR::Node *l2iNode = NULL;
-      if ((address->getOpCodeValue() == TR::aconst) && (address->getAddress() == 0))
-         {
-         l2iNode = TR::Node::create(address, TR::iconst, 0, 0);
-         }
```

Comment:
This is the "meat" of the change. Pointing this out explicitly to @0xdaryl @gita-omr @knn-k @zl-wang as it affects all codegens.

The reason we believe this change is ok is because all of our evaluators make the assumption that if we have a `TR::Address` load/store, we will have a decompression/compression sequence for the child being loaded/stored respectively. The code being reviewed here breaks that assumption and we have undefined behaviour.

Performance wise removing this piece of code should not affect performance at all, because at IL gen time storing `null` references into fields or object slots already gets optimized to an `istore` of an `iconst 0`. It is very rare that an `aconst NULL` node will reach codegen, and if it does we should _not_ try and be smart here in lower trees and do the job of the optimizer, thus breaking implicit assumptions in our evaluators.